src/gamma_inc.jl

```diff
@@ -611,3 +906,96 @@ gamma_inc(a::Float16,x::Float16,ind::Integer) = ( Float16(gamma_inc(Float64(a),F
 gamma_inc(a::Real,x::Real,ind::Integer) = (gamma_inc(float(a),float(x),ind))
 gamma_inc(a::Integer,x::Integer,ind::Integer) = gamma_inc(Float64(a),Float64(x),ind)
 gamma_inc(a::AbstractFloat,x::AbstractFloat,ind::Integer) = throw(MethodError(gamma_inc,(a,x,ind,"")))
+
+#EFFICIENT AND ACCURATE ALGORITHMS FOR THECOMPUTATION AND INVERSION OF THE INCOMPLETEGAMMA FUNCTION RATIOS by Amparo Gil, Javier Segura, Nico M. Temme
+#SIAM Journal on Scientific Computing 34(6) (2012), A2965-A2981
+# arXiv:1306.1754
+
+"""
+    gamma_inc_inv(a,p,q)
+    
+DLMF: https://dlmf.nist.gov/8.2#E4 , https://dlmf.nist.gov/8.2#E5
+Wiki: https://en.wikipedia.org/wiki/Incomplete_gamma_function
+
+gamma_inc(a,x) or (P(a,x),Q(a,x)) is the Incomplete gamma function ratio given by : ``1/\\Gamma (a) \\int_{0}^{x} e^{-t}t^{a-1} dt``
+gamma_inc_inv(a,p,q) inverts the gamma_inc function, by computing `x` given `a`,`p`,`q` in P(a,x)=p and Q(a,x)=q.
+"""
+function gamma_inc_inv(a::Float64, p::Float64, q::Float64)
+    if p < 0.5
+        pcase = true
+        porq = p
+        s = -1
+    else
+        pcase = false
+        porq = q
+        s = 1
+    end
+    m = 0
+
+    logr = (1.0/a)*( log(p) + logabsgamma(a + 1.0)[1] )
+    if logr < log(0.2*(1+a)) #small value of p
+        x0 = gamma_inc_inv_psmall(a,p)
+    elseif ((q < min(0.02, exp(-1.5*a)/gamma(a))) && (a < 10)) #small q
+        x0 = gamma_inc_inv_qsmall(a, q)
+    elseif abs(porq - 0.5) < 1.0e-05
+        m = 0
+        x0 = a - 1.0/3.0 + (8.0/405.0 + 184.0/25515.0/a) / a
+    elseif abs(a-1.0) < 1.0e-4
+        m = 0
+        x0 = pcase ? -log1p(-p) : -log(q)
+    elseif a < 1.0 # small value of a
+        x0 = gamma_inc_inv_asmall(a, p, q, pcase)
+    else    #large a
+        m = 1
+        (x0,eta) = gamma_inc_inv_alarge(a,porq,s)
+    end
+
+    t = 1
+    x = x0
+    n = 1
+    logabsgam = logabsgamma(a)[1]
+    #Newton-like higher order iteration with upper limit as 15.
+    while t > 1.0e-15 && n < 15
+        x = x0
+        x² = x*x
+        if m==0
+            dlnr = (1.0-a)*log(x) + x + logabsgam
+            if dlnr > log(floatmax(Float64)/1000.0)
+                n=20
+            else
+                r = exp(dlnr)
+            end
+        else
+            y = eta
+            fp = -sqrt(a/(2*pi))*exp(-0.5*a*y*y)/gammax(a)
```

Comment:
this only needs to be calculated once: so it could be returned as part of `gamma_inc_inv_alarge` instead of `eta`